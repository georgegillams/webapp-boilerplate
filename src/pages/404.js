import React from 'react';

import CSSHack from 'gg-webapp/components/CSSHack';
import LegacyCommonLayout from 'gg-webapp/components/LegacyCommonLayout';
import NotFound from 'containers/common/NotFound';

const Page = props => {
  return (
    <LegacyCommonLayout>
      <CSSHack pageName="404" />
      <NotFound {...props}></NotFound>
    </LegacyCommonLayout>
  );
};

Page.propTypes = {};

export default Page;
