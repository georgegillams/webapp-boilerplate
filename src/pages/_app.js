import React from 'react';

import { Provider } from 'react-redux';
import Head from 'next/head';
import App from 'next/app';
import Consent from 'containers/common/Consent';
import Analytics from 'containers/common/Analytics';
import Authenticator from 'containers/common/Authenticator';
import Konami from 'containers/common/Konami';

import withReduxStore from 'utils/common/redux/with-redux-store';
import appConfig from 'helpers/appConfig';
import Navigation from 'containers/NavigationBarWrapper';
import AppWrapper from 'components/common/AppWrapper';
import { enableES5 } from 'immer';

import 'global-styles.scss';

// Ensures that `immer` will work inside Internet Explorer
enableES5();

class Srr extends App {
  render() {
    const { Component, pageProps, reduxStore } = this.props;
    return (
      <React.StrictMode>
        <Head>
          <title>{appConfig.projectTitle}</title>
          {appConfig.app.head.meta.map(m => (
            <meta key={m.property} name={m.property} content={m.content} />
          ))}
        </Head>

        <Provider store={reduxStore}>
          <AppWrapper>
            <Navigation />
            <Konami />
            <Consent />
            <Analytics />
            <Authenticator />
            <Component {...pageProps} />
          </AppWrapper>
        </Provider>
      </React.StrictMode>
    );
  }
}

export default withReduxStore(Srr);
