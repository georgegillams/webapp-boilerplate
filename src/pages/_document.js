import React from 'react';
import Document, { Html, Head, Main, NextScript } from 'next/document';
import appConfig from 'helpers/appConfig';
import { ServerStyleSheet } from 'styled-components';
import JSFeatureDetector, { NO_JS_CLASSNAME } from '@george-gillams/components/js-feature-detector';

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const sheet = new ServerStyleSheet();
    const originalRenderPage = ctx.renderPage;

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: App => props => sheet.collectStyles(<App {...props} />),
        });

      const initialProps = await Document.getInitialProps(ctx);
      return {
        ...initialProps,
        styles: [initialProps.styles, sheet.getStyleElement()],
      };
    } finally {
      sheet.seal();
    }
  }

  render() {
    return (
      <Html lang="en-GB" className={NO_JS_CLASSNAME}>
        <Head>
          <link rel="apple-touch-icon" sizes="180x180" href="/static/favicon/favicon-180x180.png" />

          <link rel="icon" type="image/png" sizes="16x16" href="/static/favicon/favicon-16x16.png" />
          <link rel="icon" type="image/png" sizes="32x32" href="/static/favicon/favicon-32x32.png" />
          <link rel="icon" type="image/png" sizes="64x64" href="/static/favicon/favicon-64x64.png" />
          <link rel="icon" type="image/png" sizes="512x512" href="/static/favicon/favicon-512x512.png" />

          <link rel="icon" href="/static/favicon/favicon.ico" />

          <link rel="manifest" href="/static/favicon/site.webmanifest" />

          <link rel="mask-icon" href="/static/favicon/favicon.svg" color={appConfig.themeColor} />
          <link rel="shortcut icon" href="/static/favicon/favicon.png" />

          <meta name="msapplication-config" content="/static/favicon/browserconfig.xml" />
          {this.props.styleTags}
        </Head>

        <body>
          <JSFeatureDetector />
          {/* Fixes FOUC in Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1404468 */}
          <script>0</script>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;
