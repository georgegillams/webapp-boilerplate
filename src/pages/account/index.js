import React from 'react';

import Account from 'containers/Account';
import CommonLayout from 'components/CommonLayout';
import Head from 'next/head';
import appConfig from 'helpers/appConfig';

export class Page extends React.PureComponent {
  render() {
    return (
      <CommonLayout>
        <Head>
          <title>Account - {appConfig.projectTitle}</title>
        </Head>
        <Account {...this.props} />;
      </CommonLayout>
    );
  }
}

Page.propTypes = {};

Page.getInitialProps = async () => ({
  namespacesRequired: ['common', 'banner', 'features'],
});

export default Page;
