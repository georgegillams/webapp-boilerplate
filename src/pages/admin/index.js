import React from 'react';

import Admin from 'containers/Admin/Navigation';
import CommonLayout from 'components/common/CommonLayout';

const Page = props => {
  return (
    <CommonLayout>
      <Admin {...props} />
    </CommonLayout>
  );
};

Page.propTypes = {};

export default Page;
