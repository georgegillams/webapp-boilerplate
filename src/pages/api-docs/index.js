import React from 'react';

import ApiDocs from 'containers/ApiDocs';
import CommonLayout from 'components/common/CommonLayout';

const Page = props => {
  return (
    <CommonLayout>
      <ApiDocs {...props} />
    </CommonLayout>
  );
};

Page.propTypes = {};

export default Page;
