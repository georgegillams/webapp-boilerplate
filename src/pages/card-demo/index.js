import React from 'react';

import CSSHack from 'components/common/CSSHack';
import CardDemo from 'containers/CardDemo';
import CommonLayout from 'components/common/CommonLayout';

const Page = props => {
  return (
    <CommonLayout>
      <CSSHack pageName="card-demo" />
      <CardDemo {...props} />
    </CommonLayout>
  );
};

Page.propTypes = {};

export default Page;
