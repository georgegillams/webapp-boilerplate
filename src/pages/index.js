import React from 'react';

import Home from 'containers/HomeBP';
import CommonLayout from 'components/CommonLayout';

const Page = props => {
  return (
    <CommonLayout>
      <Home {...props} />
    </CommonLayout>
  );
};

Page.propTypes = {};

export default Page;
