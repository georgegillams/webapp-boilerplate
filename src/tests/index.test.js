import React from 'react';
import { act } from 'react-dom/test-utils';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'utils/configure-store';

import { IndexPage } from '../pages/index';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({});
  });

  it('should render and match the snapshot', () => {
    let props;

    act(() => {
      props = IndexPage.getInitialProps();
    });

    const { container } = render(
      <Provider store={store}>
        <IndexPage {...props} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
