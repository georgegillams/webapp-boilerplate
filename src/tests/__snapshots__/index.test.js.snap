// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render correctly 1`] = `
<div>
  <main>
    <div
      style="background: black;"
    >
      <div
        class="css-1hvjfqy-Container e1vuap0q0"
        id="exampleGetApi"
      >
        <button
          class="button__outer button__outer--regular button__outer--no-bouncy button__outer--large"
          type="button"
        >
          Get showcases
        </button>
        <div
          class="css-1bqe7xb-ShowcasesRow e1vuap0q1"
        />
      </div>
    </div>
  </main>
  <footer
    class="footer__container"
    id="footer"
  >
    <div
      class="logo__container footer__logo logo__container--centered logo__container--no-padding"
    >
      <a
        href="/"
      >
        <h1
          aria-label="Home page"
          class="logo__heading logo__heading--no-padding"
          role="button"
        >
          LOGO
        </h1>
      </a>
    </div>
    <div
      class="footer__tech"
    >
      <div
        class="typography__main typography--light typography--no-padding tech-specs__container"
      >
        Built in
        <a
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="React"
            class="tech-specs__icon tech-specs__icon--light-inverted"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        and
        <a
          href="https://redux.js.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Redux"
            class="tech-specs__icon tech-specs__icon--light-inverted"
            height="5"
            src="IMAGE_MOCK"
            width="5"
          />
        </a>
        Hosted on
        <a
          href="https://aws.amazon.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Amazon Web Services"
            class="tech-specs__icon tech-specs__icon--light-inverted"
            height="5"
            src="IMAGE_MOCK"
            style="margin-top: .8rem; max-width: 2.9rem;"
            width="8.28"
          />
        </a>
      </div>
    </div>
  </footer>
</div>
`;
